src/sync/browser.js

```diff
@@ -0,0 +1,105 @@
+import PushManagerFactory from './PushManager';
+import { PUSH_DISCONNECT, PUSH_CONNECT } from './constants';
+import FullProducerFactory from '../producer';
+import PartialProducerFactory from '../producer/browser/Partial';
+import { matching } from '../utils/key/factory';
+import { forOwn } from '../utils/lang';
+import logFactory from '../utils/logger';
+const log = logFactory('splitio-sync:sync-manager');
+
+/**
+ * Factory of sync manager for browser
+ *
+ * @param context main client context
+ */
+export default function BrowserSyncManagerFactory(mainContext) {
+
+  const contexts = {};
+  let pushManager;
+
+  function startPolling() {
+    log.info('PUSH down or disconnected. Starting periodic fetch of data.');
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
```

Comment:
Do you expect to call this without an actual producer on the context or just being cautious? When do we expect it not to have a producer? 
Can't we use the context returned promise so we kick off polling as soon as the producer is assigned? Or what would happen if you called for start polling but you had no producer and don't schedule using the promise?



Question applies to all gets here, so if one is different, I'd be interesting in knowing it as well.

Reply:
I think we have this pattern in other pieces of the code so let's have a chat about it tomorrow to make sure we're properly in sync about what and specially why I'm asking.

Reply:
There is always a producer in all contexts when calling `startPolling`, `stopPollingAndSyncAll` and `syncAll`. This is ensure by assigning the producer to the context, before creating the PushManager.
So we can remove the check `if (producer && ...`

Reply:
So why do you also pass the true value in the flagCheck param there then?

Reply:
yes. it is not needed.